### PR TITLE
fix: useCheckboxState alongside having a value="" attribute

### PR DESCRIPTION
**Pull request by gnapse**
Fixes https://github.com/ariakit/ariakit/issues/2164

We found that the problem was that, when the `defaultValue` is not initialized as an empty array, AND you also specify a `value="whatever"` attribute, the handler to change the state was not working properly. It should check if the current checkbox's value is the one already selected, and return `false` instead of returning the value. That is what makes it actually switch back to unchecked.

We also added a test, and tried to follow the existing conventions, but let us know if there's something to improve on this regard.

Co-authored alongside @pauloslund as part of our team hackathon to contribute back to open source on projects on which we rely to build our products at Doist.

### CODE REVIEW

File: packages/ariakit/src/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -60,10 +60,16 @@ function isNativeCheckbox(tagName?: string, type?: string) {
  * ```
  */
 export const useCheckbox = createHook<CheckboxOptions>(
-  ({ state, value, checked: checkedProp, defaultChecked, ...props }) => {
+  ({
+    state,
+    value: attributeValue,
```
Comment (pauloslund) [inline packages/ariakit/src/checkbox/checkbox.tsx:65]: When debugging this issue, we ran into confusion with the different "values" within this hook. We've updated the naming here to hopefully be more clear that this is the "value" attribute on the element itself.

This was done in a separate commit from the fix, so if you would like to revert this change we can.

**Review by Guria — APPROVED** [review]

**Review by diegohaz — APPROVED** [review]

**Comment by diegohaz** [comment]
Thank you so much for working on this, @gnapse and @pauloslund.
